pkg/controller/cluster_pools_pull_secret_provider/cluster_pools_pull_secret_provider_test.go

```diff
@@ -0,0 +1,249 @@
+					return err
+				}
+				expected := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: "ns",
+						Name:      "pull-secret",
+						Labels:    map[string]string{api.DPTPRequesterLabel: ControllerName},
+					},
+					Data:      map[string][]byte{corev1.DockerConfigJsonKey: []byte("abc")},
+					Type:      corev1.SecretTypeDockerConfigJson,
+					Immutable: &im,
+				}
+				if diff := cmp.Diff(expected, actual, testhelper.RuntimeObjectIgnoreRvTypeMeta); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name:   "the pool doesn't exist",
+			nn:     types.NamespacedName{Namespace: "ns", Name: "pool"},
+			client: fakeclient.NewFakeClient(srcSecret.DeepCopy()),
+			verify: func(client ctrlruntimeclient.Client) error {
+				if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: "ns", Name: "pull-secret"}, &corev1.Secret{}); !kerrors.IsNotFound(err) {
+					return fmt.Errorf("expected not found error did not occur")
+				}
+				return nil
+			},
+		},
+		{
+			name:     "the src secret is not there",
+			nn:       types.NamespacedName{Namespace: "ns", Name: "pool"},
+			client:   fakeclient.NewFakeClient(pool.DeepCopy()),
+			expected: fmt.Errorf("failed to get the secret pull-secret in namespace ci-cluster-pool: %w", fmt.Errorf("secrets \"pull-secret\" not found")),
+			verify: func(client ctrlruntimeclient.Client) error {
+				if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: "ns", Name: "pull-secret"}, &corev1.Secret{}); !kerrors.IsNotFound(err) {
+					return fmt.Errorf("expected not found error did not occur")
+				}
+				return nil
+			},
+		},
+		{
+			name:   "the pool does not use the pull secret",
+			nn:     types.NamespacedName{Namespace: "ns", Name: "pool"},
+			client: fakeclient.NewFakeClient(srcSecret.DeepCopy(), poolWithAnotherSecret.DeepCopy()),
+			verify: func(client ctrlruntimeclient.Client) error {
+				if err := client.Get(context.TODO(), ctrlruntimeclient.ObjectKey{Namespace: "ns", Name: "pull-secret"}, &corev1.Secret{}); !kerrors.IsNotFound(err) {
+					return fmt.Errorf("expected not found error did not occur")
+				}
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			// Needed so the racedetector tells us if we accidentally re-use global state, e.G. by not deepcopying
+			t.Parallel()
+			log := logrus.NewEntry(logrus.StandardLogger())
+			logrus.SetLevel(logrus.TraceLevel)
+			r := &reconciler{
+				log:                       log,
+				client:                    tc.client,
+				sourcePullSecretNamespace: "ci-cluster-pool",
+				sourcePullSecretName:      "pull-secret",
+			}
+			request := reconcile.Request{NamespacedName: tc.nn}
+			actual := r.reconcile(context.Background(), request, r.log)
```

Comment:
Nit: Mind calling `Reconcile` instead of `reconcile` here to make this fail if someone optimizes the error return away there as previously suggested?

Reply:
Updated.